control-plane/go.mod

```diff
@@ -164,4 +164,4 @@ require (
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-go 1.20
+go 1.21
```

Comment:
Does 1.22 break?

Reply:
I wouldn't expect it to. The `go <version>` directive here specifies the minimum Go version required to run the application, not necessarily the version that it will be built with since that's determined by what version of `go` is installed when you're building.

https://go.dev/ref/mod#go-mod-file-go

Reply:
Also un-bumped go version, should be done in a separate PR